.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+name: Nar
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build-macos:
+    name: Build on macos
+    runs-on: macos-11
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install build dependencies
+      run: |
+        brew update
+        brew install automake
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+    - name: Build and test the package
+      run: mvn test
+    - name: Store the build arctifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: macos-build
+        path: target/nar/louis-*
+
+  build-windows:
+    name: Build for Windows
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+    - name: Install build dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install texinfo libc6-dev-i386 mingw-w64
+    - name: Build and test the package
+      run: mvn test -Pcross-compile -Dhost.os=w64-mingw32 -Dos.arch=x86_64
+    - name: Store the build arctifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: windows-build
+        path: target/nar/louis-*
+
+  nar:
+    name: Build for linux and package the nar
```

Comment:
Could we use a single job for `build-windows` and `nar`? I guess a reason for splitting them up is to speed up the build. But then you also could have left the linux build in its own job.

Reply:
All the build jobs could be separate. I just put the linux and the deploy jobs into one out of laziness and because I didn't want to copy the linux artifacts for no good reason

Reply:
> for no good reason

A reason could be to speed up the build.

Reply:
Copying artifacts takes time :-(

Reply:
But I'm fine either way

Reply:
Linux build is a separte job now